### DISCUSSION TITLE
Display Posts/Content in Latest Posts

### DISCUSSION THREAD

**Discussion by bloodyeyesblackblood** [11ty/eleventy#2549] [Uncategorized]
I created a Design and now recreating it in 11ty~
![image](https://user-images.githubusercontent.com/108025342/188285103-41461911-9b2f-487e-a683-729f34d1a6b2.png)

Im struggling so much, since this is really new to me.. so I did.. post.data.description but it just displays the current Description and not the post itself, what should I use for this? and also how can I limit it to 20 words?

Currently it looks like this...
![image](https://user-images.githubusercontent.com/108025342/188285172-7eea1f8e-a239-4191-8a4f-7d5e8596d7e6.png)

Edit: didnt found a similiar post, if there is one, link it please.

**Comment by pdehaan** [comment]
@bloodyeyesblackblood Do you have a public repo or something?
Not sure how much I can help w/ just screenshots.

If you're using liquidjs, there is a [`truncatewords`](https://liquidjs.com/filters/truncatewords.html) filter which will truncate by words instead of just character limits.

**Reply by bloodyeyesblackblood** [reply]
I downloaded a template from someone, it uses:

---
layout: layouts/home.njk
eleventyNavigation:
  key: Home
  order: 1
---
{% set maxPosts = collections.posts.length | min(6) %}

{% set postslist = collections.posts | head(-6) %}
{% set postslistCounter = collections.posts | length %}
{% include "postslist.njk" %}

<p>More posts can be found in <a href="{{ '/posts/' | url }}">the archive</a>.</p>

**Reply by pdehaan** [reply]
Here was my proof of concept showing looping over a collection and displaying a list of posts and their `url` and `data.description` and truncating to 4 words: https://github.com/pdehaan/11ty-2549

**Reply by bloodyeyesblackblood** [reply]
whats better for github static sites? liquid or md? what the difference? also i need to get the text in the post (content), not the site description :(

**Reply by pdehaan** [reply]
I don't think there is a "better". Each engine has pros and cons. By default .md templates will also process liquid tags (which can be useful), but indenting and formatting can make .md files a bit tricky sometimes.
I generally prefer using (in no order): [LiquidJS](https://www.11ty.dev/docs/languages/liquid/), [Nunjucks](https://www.11ty.dev/docs/languages/nunjucks/), or [JavaScript/Eleventy](https://www.11ty.dev/docs/languages/javascript/) templating.

As for getting the template content, I believe you can use `{{ page.templateContent }}`. Per https://www.11ty.dev/docs/collections/#collection-item-data-structure

> _`templateContent`: the rendered content of this template. This does not include layout wrappers._

If you're trying to pull excerpts, note that the docs actually have a section on that very thing: https://www.11ty.dev/docs/data-frontmatter-customize/.

**Reply by bloodyeyesblackblood** [reply]
> I don't think there is a "better". Each engine has pros and cons. By default .md templates will also process liquid tags (which can be useful), but indenting and formatting can make .md files a bit tricky sometimes. I generally prefer using (in no order): [LiquidJS](https://www.11ty.dev/docs/languages/liquid/), [Nunjucks](https://www.11ty.dev/docs/languages/nunjucks/), or [JavaScript/Eleventy](https://www.11ty.dev/docs/languages/javascript/) templating.
> 
> As for getting the template content, I believe you can use `{{ page.templateContent }}`. Per https://www.11ty.dev/docs/collections/#collection-item-data-structure
> 
> > _`templateContent`: the rendered content of this template. This does not include layout wrappers._
> 
> If you're trying to pull excerpts, note that the docs actually have a section on that very thing: https://www.11ty.dev/docs/data-frontmatter-customize/.

thanks after your comment i began to look trough some posts with template content {{ post.templateContent | safe }} did help indeed, now I need to find how to limit the text to like 20-40 words.

since excerpt didnt work that well or im really way to dumb to use it

(I created a whole new project, because the base template was to big and messy for me, so all from scratch to learn it kinda myself, might upload it in the next hours)

**Reply by pdehaan** [reply]
Since you're using liquidjs, I'd probably use a combination of [`strip_html`](https://liquidjs.com/filters/strip_html.html) and [`truncatewords`](https://liquidjs.com/filters/truncatewords.html) filters. Possibly something like this
```liquid
{{ page.templateContent | strip_html | truncatewords: 20 }}
```

**Warning:** Be careful about using `page` as a local/loop variable name, since Eleventy already supplies a `page` variable, see https://www.11ty.dev/docs/data-eleventy-supplied/#page-variable.

I find excerpts pretty hard to do, since if the beginning of your `templateContent` has `<h1>` tags or something, it might not always be great excerpts. But that all depends on the input/layout. But sometimes better than the alternative of manually setting an excerpt for each page (depending on how big your site is).

**Reply by bloodyeyesblackblood** [reply]
nope im still using njk
https://github.com/bloodyeyesblackblood/bloodyeyes

since i cant push it at the time, cause idk how git works (console says current one is way top behind), here a new branch: https://github.com/bloodyeyesblackblood/bloodyeyes/tree/bloodyeyes

**Reply by pdehaan** [reply]
Ah, should be doable w/ Nunjucks, although you would probably need use [`striptags`](https://mozilla.github.io/nunjucks/templating.html#striptags-value-preserve_linebreaks) (Nunjucks' version of `strip_html` filter) and  do a rough `truncatewords` implementation yourself.

```js
  eleventyConfig.addFilter("truncatewords", function (str = "", limit = 30) {
    return str.toString()
      .trim()
      .split(/\s+/g, limit)
      .join(" ") + "&hellip;";
  });
```

### USAGE

```njk
---
title: Homepage
layout: layouts/base.njk
---
<h1>w</h1>
<ul>
{% for post in collections.posts %}
	<li>
		<a href="{{post.url}}">{{ post.data.title }}</a>
		<p>{{ post.templateContent | striptags(true) | truncatewords(15) | safe }}</p>
	</li>
{% endfor %}
</ul>
```
